apps/circuit-compiler/src/app/actions/index.ts

```diff
@@ -162,14 +96,96 @@ export const generateProof = async (plugin: CircomPluginClient, appState: AppSta
+  const vKey = await snarkjs.zKey.exportVerificationKey(zkeyFinal)
+
+  await exportVerificationFiles(plugin, appState, vKey, GROTH16_VERIFIER, zkeyFinal, 'groth16')
+  dispatch({ type: 'SET_ZKEY', payload: zkeyFinal })
+  dispatch({ type: 'SET_VERIFICATION_KEY', payload: vKey })
+}
+
+async function setupPlonk(plugin: CircomPluginClient, appState: AppState, dispatch, r1cs, ptauFinal, zkeyFinal) {
+  await snarkjs.plonk.setup(r1cs, ptauFinal, zkeyFinal, zkLogger(plugin, dispatch, 'SET_SETUP_EXPORT_FEEDBACK'))
+  const vKey = await snarkjs.zKey.exportVerificationKey(zkeyFinal)
+
+  await exportVerificationFiles(plugin, appState, vKey, PLONK_VERIFIER, zkeyFinal, 'plonk')
+  dispatch({ type: 'SET_ZKEY', payload: zkeyFinal })
+  dispatch({ type: 'SET_VERIFICATION_KEY', payload: vKey })
+}
+
+async function proveGroth16(plugin: CircomPluginClient, appState: AppState, dispatch, wtns) {
+  const { proof, publicSignals } = await snarkjs.groth16.prove(appState.zKey, wtns, zkLogger(plugin, dispatch, 'SET_PROOF_FEEDBACK'))
+  const verified = await snarkjs.groth16.verify(appState.verificationKey, publicSignals, proof)
+
+  await writeProofFiles(plugin, appState, proof, 'groth16', verified)
+}
+
+async function provePlonk(plugin: CircomPluginClient, appState: AppState, dispatch, wtns) {
+  const { proof, publicSignals } = await snarkjs.plonk.prove(appState.zKey, wtns, zkLogger(plugin, dispatch, 'SET_PROOF_FEEDBACK'))
+  const verified = await snarkjs.plonk.verify(appState.verificationKey, publicSignals, proof)
+
+  await writeProofFiles(plugin, appState, proof, 'plonk', verified)
+}
+
+async function exportVerificationFiles(plugin, appState, vKey, verifierTemplate, zkeyFinal, scheme) {
+  if (appState.exportVerificationKey) {
+    await writeFile(plugin, `${extractParentFromKey(appState.filePath)}/${scheme}/zk/keys/verification_key.json`, JSON.stringify(vKey, null, 2))
+  }
+
+  if (appState.exportVerificationContract) {
+    const solidityContract = await snarkjs.zKey.exportSolidityVerifier(zkeyFinal, verifierTemplate)
+    await writeFile(plugin, `${extractParentFromKey(appState.filePath)}/${scheme}/zk/verifier.sol`, solidityContract)
+  }
+}
+
+async function writeProofFiles(plugin, appState, proof, scheme, verified) {
+  const proofPath = `${extractParentFromKey(appState.filePath)}/${scheme}/zk/proof.json`
+  await writeFile(plugin, proofPath, JSON.stringify(proof, null, 2))
+
+  if (verified) {
+    trackEvent(plugin, 'proof_verified', proofPath, "notification")
+  } else {
+    trackEvent(plugin, 'proof_failed', proofPath, "notification")
+  }
+}
+
+function trackEvent(plugin, eventCategory, action, label) {
+  plugin._paq.push(['trackEvent', eventCategory, action, label])
+}
+
+async function readFileAsUint8Array(plugin: CircomPluginClient, path: string): Promise<Uint8Array> {
+  const data = await plugin.call('fileManager', 'readFile', path)
```

Comment:
Missing parameter to fileManager plugin call `'opt?'` parameter. `"{ encoding: null }"`. The type definition for `fileManager` is not updated for this additional option. You can use // @ts-ignore to ignore the error.